go/private/sdk.bzl

```
@@ -277,6 +283,10 @@ def _detect_host_platform(ctx):
 
     elif ctx.os.name.startswith("windows"):
         goos, goarch = "windows", "amd64"
+        if _get_env_var(ctx, "PROCESSOR_ARCHITECTURE") == "ARM64":
+            goarch = "arm64"
+        elif _get_env_var(ctx, "PROCESSOR_ARCHITEW6432") == "ARM64":
+            goarch = "arm64"
```

```suggestion
        goos = "windows"
        # TODO: use ctx.os.arch after minimal version of Bazel becomes 5.1.
        if ctx.os.environ.get("PROCESSOR_ARCHITECTURE") == "ARM64" or ctx.os.environ.get("PROCESSOR_ARCHITEW6432") == "ARM64":
            goarch = "arm64"
        else:
            goarch = "amd64"
```

* Avoid assigning `goarch` to "amd64" and immediately override it.
* Don't implement something that `dict` already has.